### PR TITLE
chore: add workspaces

### PR DESCRIPTION
**Pull request by jakubkoci**
This PR introduces Yarn Workspaces with two packages: frontend and backend. The frontend package will contain the UI built with Next.js, and the backend is just our current Node.js service built with Express. Because Next.js runs its own server by default, we must figure out how to run them all together.

A. Run just Next.js and implement our backend as API routes inside the Next.js frameworks.
- pros: one Node.js server handling everything
- cons: no separate of concerns, requires significant changes in our project

B. Run two Node.js servers on a different port
- pros: better separation of concerns (API and UI are independent services), we can keep our backend code untouched as is and just add frontend package
- cons: running two separate Node.js services, either each in one docker container or one container for both exposing two ports for each service

I already dismissed the first option because it would require significant changes, and I feel that running two services shouldn't be so difficult, it might be beneficial long-term. Or we can always move our endpoint to the frontend Next.js as API routes.

I only set up the project in a way that it should run just Node.js backend as we have done it so far. Later, we can figure out how to run the frontend part.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,10 +9,15 @@ RUN apt-get update && apt-get install -y \
   gcc \
   make \
   python3
+
+RUN yarn config set strict-ssl false
+ENV NODE_TLS_REJECT_UNAUTHORIZED=0
```
Comment (niall-shaw) [inline Dockerfile:14]: FYI - I think this is necessary internally within ABSA, not sure if we should include in this repo?

File: packages/backend/package.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{
+  "name": "@ssi-trust-registry/backend",
+  "version": "1.0.0",
+  "main": "index.js",
+  "license": "MIT",
+  "scripts": {
+    "build": "rm -rf build && tsc",
+    "format:check": "prettier --check .",
+    "format:update": "prettier --write .",
+    "lint": "eslint . --ignore-path ../../.gitignore",
```
[the remark this answers is not in the document]
Comment (niall-shaw) [reply packages/backend/package.json:10]: Perhaps we should use a linter outside of this package to lint everything?

[the remark this answers is not in the document]
Comment (jakubkoci) [reply packages/backend/package.json:10]: Good point. I'll double-check, but the idea for now is just to make it work somehow. Then we should move as much as possible to the root. It's also my first time with worskpaces so I'm still not sure what should be in package and what in root. For example, in relation to estlint, it's a question if frontend package wouldn't use a bit different/additional rules for the frontend/React stuff.

[the remark this answers is not in the document]
Comment (jakubkoci) [reply packages/backend/package.json:10]: Yes, we should either create a specific gitignore for backend package or use one eslint config file for everything how AFJ does it. On the other hand, our frontend package seems quite specific but packages in AFJ are usually all of a similar kind.

**Review by niall-shaw — APPROVED** [review]

**Pull request merged**